lib/improver/tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py

```diff
@@ -0,0 +1,558 @@
+        data[time_index][lat_index][lon_index] = 0
+
+    cube = Cube(data, standard_name="precipitation_amount",
+                units="kg m^-2")
+
+    if source_realizations is not None:
+        if isinstance(source_realizations, list):
+            cube.attributes.update(
+                {'source_realizations': source_realizations})
+
+    tunit = Unit("hours since 1970-01-01 00:00:00", "gregorian")
+    time_points = [402192.5 + _ for _ in range(num_time_points)]
+    cube.add_dim_coord(DimCoord(time_points,
+                                standard_name="time", units=tunit), 0)
+
+    step_size = 2000
+    y_points = np.arange(0, step_size*num_grid_points, step_size)
+    cube.add_dim_coord(
+        DimCoord(
+            y_points,
+            'projection_y_coordinate',
+            units='m'
+        ),
+        1
+    )
+
+    x_points = np.arange(-50000, (step_size*num_grid_points)-50000, step_size)
+    cube.add_dim_coord(
+        DimCoord(
+            x_points,
+            'projection_x_coordinate',
+            units='m'
+        ),
+        2
+    )
+
+    return cube
+
+
+def set_up_cube_lat_long(zero_point_indices=((0, 7, 7),), num_time_points=1,
+                         num_grid_points=16):
+    """Set up a lat-long coord cube."""
+    data = np.ones((num_time_points, num_grid_points, num_grid_points))
+    for time_index, lat_index, lon_index in zero_point_indices:
+        data[time_index][lat_index][lon_index] = 0
+    cube = Cube(data, standard_name="precipitation_amount",
+                units="kg m^-2")
+    tunit = Unit("hours since 1970-01-01 00:00:00", "gregorian")
+    time_points = [402192.5 + _ for _ in range(num_time_points)]
+    cube.add_aux_coord(
+        AuxCoord(time_points, "time", units=tunit), 0)
+    cube.add_dim_coord(
+        DimCoord(np.linspace(0.0, float(num_grid_points - 1),
+                             num_grid_points),
+                 'latitude',
+                 units='degrees'),
+        1
+    )
+    cube.add_dim_coord(
+        DimCoord(np.linspace(0.0, float(num_grid_points - 1),
+                             num_grid_points),
+                 'longitude',
+                 units='degrees'),
+        2
+    )
+    return cube
+
+
+class Test__init__(IrisTest):
+
+    """Test the __init__ method of NeighbourhoodProcessing"""
+
+    def test_radii_varying_with_lead_time_mismatch(self):
+        """
+        Test that the desired error message is raised, if there is a mismatch
+        between the number of radii and the number of lead times.
+        """
+        radii = [10000, 20000, 30000]
+        lead_times = [2, 3]
+        msg = "There is a mismatch in the number of radii"
+        with self.assertRaisesRegexp(ValueError, msg):
+            neighbourhood_method = CircularNeighbourhood()
+            NBHood(neighbourhood_method, radii, lead_times=lead_times)
+
+
+class Test__repr__(IrisTest):
+
+    """Test the repr method."""
+
+    def test_basic(self):
```

Comment:
I feel like we should maybe have two tests here to test both branches of the if statement.

Reply:
I've removed the if statement from the `repr` method.

Reply:
I've added this back in, as it is necessary depending upon how `BaseNeighbourhoodProcessing` is called. If it's called directly, as done in the unit tests, then the `neighbourhood_method` could be a callable object, however, if `BaseNeighbourhoodProcessing` is used as a Base Class for e.g. `NeighbourhoodProcessing` then the `neighbourhood_method` will be just `circular` or `square` as the conversion from e.g `circular` into an object is done within the subclasses.

Reply:
And two tests added for a callable case and a non-callable case